### PR TITLE
Add support for optional parameters in a test method

### PR DESCRIPTION
**Pull request by hughbe**
I seem to remember seeing an issue in the repo requesting something like this but can't find it now!

Also: http://stackoverflow.com/questions/14818439/xunit-test-engines-inlinedataattribute-optional-method-parameters

/cc @bradwilson

### CODE REVIEW

**Comment by hughbe** [comment]
Hmm, how odd that the tests are failing: using TestDriven.NET to run tests (as per the recommended contributor guidelines), I get all passes
tests

@bradwilson is there some sort of weird CI stuff going on? Could you take a look when you have time :)

**Comment by hughbe** [comment]
Okay, so it turns out xunit has two different runners of theory tests! I had only managed to test one of them from TestDriven.net, but the CI tests a different one.

I've fixed this and removed some code that isn't necessary at all

**Comment by hughbe** [comment]
@bradwilson can you take a look if you have time - I keep coming up with places I wanna use this in my tests, thanks!

File: test/test.xunit.execution/Acceptance/Xunit2TheoryAcceptanceTests.cs [hunk 1]
```diff
@@ -14,6 +13,102 @@ public class Xunit2TheoryAcceptanceTests
...
+            [Theory]
+            [InlineData("abc", 6)]
+            public void OneOptional_TwoParameters_TwoPassed(string s, int i = 5)
+            {
+                Assert.Equal("abc", s);
+                Assert.Equal(6, i);
+            }
+
+            [Theory]
+            [InlineData()]
```
Comment (bradwilson) [inline test/test.xunit.execution/Acceptance/Xunit2TheoryAcceptanceTests.cs]: Never thought I'd see this. :-p

**Comment by bradwilson** [comment]
Merged, thanks!

**Pull request closed without merging**